### PR TITLE
[r] Publish pkgdown docs from `main`, no longer from `main-old`

### PR DESCRIPTION
**Pull request by johnkerl**
**Issue and/or context:** Counterpart to PR #1095 for issue #1041.

**Changes:** The `main-old` are now locally rendered only.

**Notes for Reviewer:**

### CODE REVIEW

File: .github/workflows/pkgdown-yaml-is-in-main-now [hunk 1]
```diff
@@ -1,3 +1,6 @@
+# Update 2023-03-10: Now that 1.0.0 approaches we are publishing docs from the main branch, no
+# longer from main-old.
+
 # Workflow derived from https://github.com/r-lib/actions/tree/master/examples
 # Need help debugging build failures? Start at https://github.com/r-lib/actions#where-to-find-help
 on:
```
Comment (eddelbuettel) [inline .github/workflows/pkgdown-yaml-is-in-main-now:6]: You can comment-out (or remove) lines 21 to 24 here.

**Review by mojaveazure — APPROVED** [review]
Do we want these docs removed entirely or built into a separate directory so that https://single-cell-data.github.io/TileDB-SOMA/ refers to `main` while something like `https://single-cell-data.github.io/TileDB-SOMA/main-old` hosts the docs for `main-old` (see `subdir` argument to [`pkgdown::deploy_to_branch()`](https://pkgdown.r-lib.org/reference/deploy_to_branch.html)?

**Comment by johnkerl** [comment]
@mojaveazure sure but I don't know how to do that.

Right now we _must_ disable this b/c any `main-old` commit will overwrite `main` docs (same URL).

Happy to take this as a follow-on PR.

**Comment by johnkerl** [comment]
(The deploy-to-branch is already, effectively, in use -- the `gh-pages` branch is used. What I don't know is how to deploy-to-subdir, nor how to get GitHub Pages to know about subdirs.)

**Pull request merged**